service/history/execution/mutable_state_builder.go

```diff
@@ -1181,6 +1181,22 @@ func (e *mutableStateBuilder) AddWorkflowExecutionTerminatedEvent(
 	if err := e.ReplicateWorkflowExecutionTerminatedEvent(firstEventID, event); err != nil {
 		return nil, err
 	}
+
+	domainName := e.GetDomainEntry().GetInfo().Name
+
+	e.logger.Info(
+		"Workflow execution terminated.",
+		tag.WorkflowDomainName(domainName),
+		tag.WorkflowID(e.GetExecutionInfo().WorkflowID),
+		tag.WorkflowRunID(e.GetExecutionInfo().RunID),
+		tag.WorkflowTerminationReason(reason),
+	)
+
+	scopeWithDomainTag := e.metricsClient.Scope(metrics.HistoryTerminateWorkflowExecutionScope).
+		Tagged(metrics.DomainTag(domainName)).
+		Tagged(metrics.WorkflowTerminationReasonTag(reason))
```

Comment:
This looks good, the only thing I'd be slightly concerned about is if there's invalid chars in the workflow reason string which cause the metric to get dropped. What do you think about a regex filter or something to just remove spaces and non-alphanumerics? 

```
re := regexp.MustCompile("[^a-zA-Z0-9]")
reason = re.ReplaceString(reason, "_") // or whatever the method is, I forget
```